src/components/layers/drawing/drawInteraction.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useRef, useState } from 'react'
+import { useMap } from '../../../contexts/mapContext/mapContext'
+import Draw from 'ol/interaction/Draw'
+import VectorSource from 'ol/source/Vector'
+import VectorLayer from 'ol/layer/Vector'
+import { Box, Button, Typography } from '@mui/material'
+import { getArrayOfAllLayers, getArrayOfVectorLayersWithoutDrawing } from '../getArrayOfLayers'
+import featuresCounter from './featuresCounter'
+
+const DrawInteraction = () => {
+    const mapRef = useRef(useMap())
+    const [disableButton, setDisableButton] = useState(false)
+    const [counterArray, setCounterArray] = useState(null)
+    const [vector_layers_without_drawing, set_vector_layers_without_drawing] = useState([])
+
+    const onDraw = (event) => {
+        const layers_without_drawing = getArrayOfVectorLayersWithoutDrawing(mapRef.current)
+        set_vector_layers_without_drawing(layers_without_drawing)
+        const geometry = event.feature.getGeometry();
+        const counter = featuresCounter(layers_without_drawing, geometry)
+        setCounterArray(counter)
+    }
+
+    const handleInteraction = () => {
+        if (!mapRef.current) return;
+        const source = getArrayOfAllLayers(mapRef.current).find(element => element.getProperties().name === 'drawLayer').getSource()
+        const draw = new Draw({
+          source: source,
+          type: "Polygon",
+        });
+
+        draw.on('drawend', onDraw);
+
+        mapRef.current.addInteraction(draw);
+        source.on('addfeature', function(evt){
+            draw.setActive(false);
+        });
+        setDisableButton(true)
+
+    }
+
+    const handleClearDraw = () => {
+        if (!mapRef.current) return;
+
+        getArrayOfAllLayers(mapRef.current).find(element => element.getProperties().name === 'drawLayer').getSource().clear();

```

Comment:
I advise creating a collection like LayerIds or LayerNames to avoid the necessity of using constants.